class-13.md

```diff
@@ -0,0 +1,138 @@
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gnupg-agent \
+    software-properties-common
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+
+sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
+
+sudo apt-get update
+
+sudo apt-get install docker-ce
+```
+
+Similar to how Git has versioned states of code, Docker images are available in a similar repository. There is a service called DockerHub as well for sharing them, similar to GitHub. You can test out a basic example by running:
+
+```bash
+docker run hello-world
+```
+
+This will try to find an image called `hello-world`, and if it can't find it locally it will search through DockerHub for it. We'll find one, and play back a success message when it is successfully built.
+
+Docker images can start with common configurations for many projects, and you can locate them in a similar manner to searching through a package manager. You can run the following command:
+
+```bash
+docker search ubuntu
+```
+
+Which will return a list of all images that contain the name `ubuntu`. When building one, you could pull and configure it to serve as the starting point of a project. You can also push your images up to DockerHub to share later.
+
+To list the images you have installed, run a list command:
+
+```bash
+docker images
+```
+
+This will return a list of all docker images on your computer. 
+
+Finally, to run the image we searched, you can run it in interactive mode with the following command:
+
+```
+docker run -it ubuntu
+```
+
+After running this, you should enter a shell inside a ubuntu VM. You can configure any dependencies in your environment from here, or if you don't have an ubuntu setup and would like to follow along with the instructions anyways, you can use this command.
+
+#### Managing Your Containers
+
+You can view active containers with the following command:
+
+```bash
+docker ps
+```
+
+You can use the `-a` switch afterwards to see only active containers, or the `-l` switch to see the last container you just created. When listing containers, you should see each result with a container id. This id is how you reference specific docker containers in future commands, sucn as starting or stopping it:
+
+##### Starting
+
+```bash
+docker start my_container_id_here
+```
+
+##### Stopping
+
+```bash
+docker stop my_container_id_here
+```
+
+In addition to starting/stopping by id, you can also use the container name (for the above example, it would be `ubuntu`).
+
+Now that we have a basic overview of docker and can start/stop containers, let's move to the next section about working with fabric.
+
+### Moving to Fabric
+
+We have reviewed Hyperledger Composer pretty extensively, now we can look at a production application that is based on Fabric specifically. It could also be implemented on composer, but we can showcase some of the differences and other ways it could be modified here.
```

Comment:
Really need to avoid ambiguous pronouns like "it":
`It could also be implemented on composer, but we can showcase some of the differences and other ways it could be modified here.`
->
`{The production application} can also be implemented on composer, but we can first showcase the differences and possible modifications for {state possible purposes for modifications}.`